app/controllers/active_storage/direct_uploads_controller.rb

```diff
@@ -4,11 +4,18 @@
 class ActiveStorage::DirectUploadsController < ActionController::Base
   def create
     blob = ActiveStorage::Blob.create_before_direct_upload!(blob_args)
-    render json: { upload_to_url: blob.service_url_for_direct_upload, signed_blob_id: blob.signed_id }
+    render json: direct_upload_json(blob)
   end
 
   private
     def blob_args
       params.require(:blob).permit(:filename, :byte_size, :checksum, :content_type, :metadata).to_h.symbolize_keys
     end
+
+    def direct_upload_json(blob)
+      blob.as_json(methods: :signed_id).merge(direct_upload: {
```

Comment:
Not technically, but it robustifies the client-side API with a complete set of attributes from the server.

Before:
<img width="819" alt="before" src="https://user-images.githubusercontent.com/5355/28747944-0a0a9764-7479-11e7-814d-aeb725643d94.png">
After:
<img width="764" alt="after" src="https://user-images.githubusercontent.com/5355/28747947-0f617fb6-7479-11e7-855c-c4f39f93377d.png">